### PR TITLE
Added security definitions along with responses

### PR DESCRIPTION
**Pull request by udit-zededa**
### My PR contains... 

- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [ ] Bug fixes (non-breaking change which fixes an issue)
- [ ] Improvements (misc. changes to existing features)
- [ ] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [ ] are not breaking changes.

### Documentation
- [ ] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [ ] My changes can not or do not need to be tested.
- [ ] My changes can and should be tested by unit and/or integration tests.
- [ ] If yes to above: I have added tests to cover my changes.
- [ ] If yes to above: I have taken care to cover edge cases in my tests.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/core/components/auth/authorization-popup-perAPI.jsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import React from "react"
+import PropTypes from "prop-types"
+
+export default class AuthorizationPopupPerAPI extends React.Component {
```
Comment (ssat-zededa) [inline src/core/components/auth/authorization-popup-perAPI.jsx:4]: This isn't really a popup is it? Given it is displayed inline and explains the API accessibility, how about we name this component simply as `APIAccessDetails`?

File: src/core/components/auth/authorization-popup-perAPI.jsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  close =() => {
```
Comment (ssat-zededa) [inline src/core/components/auth/authorization-popup-perAPI.jsx:5]: Do we need this? I don't see anything in the rendered view calling this function.

**Review by ssat-zededa — CHANGES_REQUESTED** [review]
Please check comments. Also another question:
Are we planning to retain the current trigger (lock icon?) that loads the popup in which you are displaying contents as updated in this PR?

File: src/core/components/auth/authorization-popup-perAPI.jsx (hunk 1 above)
Comment (udit-zededa) [reply src/core/components/auth/authorization-popup-perAPI.jsx:4]: Yaa sure, I will rename this component as APIAccessDetails

File: src/core/components/auth/authorization-popup-perAPI.jsx (hunk 2 above)
Comment (udit-zededa) [reply src/core/components/auth/authorization-popup-perAPI.jsx:5]: No,
I will remove it

File: src/core/components/auth/authorization-popup-perAPI.jsx (hunk 1 above)
Comment (udit-zededa) [reply src/core/components/auth/authorization-popup-perAPI.jsx]: Resolved.

File: src/core/components/auth/authorization-popup-perAPI.jsx (hunk 2 above)
Comment (udit-zededa) [reply src/core/components/auth/authorization-popup-perAPI.jsx]: Resolved.

File: src/core/components/auth/auth-item-access-details.jsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import React from "react"
+import PropTypes from "prop-types"
+import ImPropTypes from "react-immutable-proptypes"
+
+export default class AuthsAccessDetails extends React.Component {
```
Comment (ssat-zededa) [inline src/core/components/auth/auth-item-access-details.jsx:5]: Please add documentation explaining what the class does. Is there a reason the class name doesn't match the file name (I would expect camel case to hyphen conversion without drop in words).

**Review by ssat-zededa — CHANGES_REQUESTED** [review]

File: src/core/components/auth/auth-item-access-details.jsx (hunk 3 above)
Comment (udit-zededa) [reply src/core/components/auth/auth-item-access-details.jsx]: The file name changed as per suggestion.
